lib/syskit/models/dynamic_port_binding.rb

```diff
@@ -197,9 +197,24 @@ def initialize(name, component_model, *arguments, **kw_arguments)
             class BoundOutputReader < OutputReader
                 include BoundAccessor
 
+                def initialize(
+                    name, component_model, *arguments,
+                    klass: Syskit::DynamicPortBinding::BoundOutputReader, **kw_arguments
+                )
+                    super(name, component_model, *arguments, **kw_arguments)
+
+                    unless klass <= Syskit::DynamicPortBinding::BoundOutputReader
+                        raise ArgumentError,
+                              "#{klass} does not subclass " \
+                              "Syskit::DynamicPortBinding::BoundOutputReader"
+                    end
+
```

Comment:
This is Ruby. That's not your problem (specifically at this level of the usage of the API)